Create Multiple tables

danielmatama opened the pull request:
I created the owner's table.
I created the species table.
I inserted data into the owners and species tables.
I m
![Screenshot from 2023-03-31 10-32-16](https://user-images.githubusercontent.com/110289566/229057281-e770c0f7-2f9b-41da-bcc5-bf2a4f4a0cc1.png)
![Screenshot from 2023-03-31 10-37-59](https://user-images.githubusercontent.com/110289566/229057356-ff2d4aed-12d5-4264-8f9b-567f850a7056.png)
![Screenshot from 2023-03-31 10-41-13](https://user-images.githubusercontent.com/110289566/229057450-93aeb987-955c-4b79-9e3f-900e0bf23b4e.png)
odified the animals table to include species_id and owners_id columns and make them foreign keys

addod19 commented on queries.sql:
```diff
@@ -1,3 +1,120 @@
+/* How many animals are there per species? */
+SELECT COUNT(animals.name), species.name FROM animals JOIN species ON animals.species_id = species.id GROUP BY species.name;
+
+/* List all Digimon owned by Jennifer Orwell. */
+SELECT owners.full_name, animals.name AS animal_name, species.name AS species_name FROM animals JOIN owners ON owners.id = animals.owner_id JOIN species ON species.id = animals.species_id WHERE owners.full_name = 'Jennifer Orwell' AND species.name = 'Digimon';
```
- [ ] It would be better if we take a second look at the results of this query...Currently, we can not find any data as a result of the query, however, we are expecting to find a data from this query. Kindly check the screenshot to this query

HAMMAS-SALEEM replied:
**[OPTIONAL]**

- [ ] It would be nice if this query returns Gabumon because Gabumon is a Digimon and owned by Jennifer Orwell 👍 right now your query is not returning anything 😨
- [ ] After making all the changes it would be great if you take new screenshots of the queries requested by me and replace them with the old screenshots in your pull request description 👍

addod19 commented on queries.sql:
```diff
@@ -1,3 +1,120 @@
+SELECT name, escape_attempts, full_name FROM animals JOIN owners ON animals.owner_id = owners.id WHERE escape_attempts = 0 AND full_name = 'Dean Winchester';
+
+/* Who owns the most animals? */
+SELECT full_name, COUNT(animals.owner_id) as total_animals FROM animals JOIN owners ON animals.owner_id = owners.id GROUP BY owners.id ORDER BY total_animals DESC LIMIT 1;
```
- [ ] I am unable to see the results of this query in the screenshot. Kindly make the results of the screenshot visible

addod19 reviewed (changes requested):
### Hi @danielmatama, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [ ] Correct Git-flow followed :100: 
- [ ] Most queries are supported with various screesnhots :rocket: 
- [ ] Schema and data population clean and awesome :fireworks: 


## Required Changes ♻️
_Check the comments under the review._
- [ ] It would be better if we remove the author2 section since this project is not a collaborative one. Also if we can not write the features for our project, let's not leave it like `key feature3` or `new feature 1` etc. Let us try and make our repo professional

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Capitalizing all commit messages going forward :1st_place_medal: 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

danielmatama commented:
![Screenshot from 2023-03-31 12-09-19](https://user-images.githubusercontent.com/110289566/229078799-19de5aea-2f28-4258-963d-2244ae2d6490.png)

HAMMAS-SALEEM commented on README.md:
```diff
@@ -176,43 +110,16 @@ Example command:
+- LinkedIn: [LinkedIn](https://linkedin.com/in/danielmatamamwebesa)
 
 👤 **Author2**
 
```
- [ ] It would be great if you remove this Author2 since there is no collaborator in this project other than you 🤔 as suggested by the previous reviewer 👍 to make your readme file descriptive enough 😊

HAMMAS-SALEEM commented on README.md:
```diff
@@ -226,9 +133,7 @@ Example:
-- [ ] **[new_feature_1]**
+- [ ] **[owners'table]**
 - [ ] **[new_feature_2]**
 - [ ] **[new_feature_3]**
```
- [ ] As suggested by the previous reviewer that it would be nice if you don't leave your features like **new_feature3** or **new_feature2** because these are just placeholders which are needed to be removed 👍 Kindly remove these placeholders and add some features here to make your readme more descriptive 👍

HAMMAS-SALEEM commented on README.md:
```diff
@@ -63,72 +23,50 @@ After you're finished please remove all the comments and instructions!
+- **[animal tables]**
+- **[animal data]**
 - **[key_feature_3]**
 
```
- [ ] As suggested by the previous reviewer that it would be nice if you don't leave your features like key_feature3 because these are just placeholders which are needed to be removed 👍 Kindly remove these placeholders and add some features here to make your readme more descriptive 👍

HAMMAS-SALEEM commented on queries.sql:
```diff
@@ -1,3 +1,120 @@
+
+/* List of owners and their animals, remember to include those that don't own any animal. */
+SELECT name, full_name FROM animals FULL JOIN owners ON animals.owner_id = owners.id;
+
+/* How many animals are there per species? */
+SELECT COUNT(animals.name), species.name FROM animals JOIN species ON animals.species_id = species.id GROUP BY species.name;
+
```
**[OPTIONAL]**

- [ ] It would be better if this query returns a total of 10 entries because if you see in the screenshot you provided there are only a total of 9 entries 🤔 You have 10 animals in your database now 🤔

HAMMAS-SALEEM commented on README.md:
```diff
@@ -63,72 +23,50 @@ After you're finished please remove all the comments and instructions!
+- **[animal data]**
 - **[key_feature_3]**
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
```
![image](https://user-images.githubusercontent.com/93039359/229088855-1ddb6360-bfbe-4beb-b5ef-2e082604cba3.png)

**[OPTIONAL]**

- [ ] It would be nice if you add the following sections in your readme file to make your readme more professional and descriptive 👍

HAMMAS-SALEEM reviewed (changes requested):
# Required Changes :recycle:
![almost-there](https://user-images.githubusercontent.com/65860678/170491251-b1add8ba-b18f-4648-a893-20e510d957fd.gif)

## Hey @danielmatama :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requested changes, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

danielmatama commented:
I updated the animals' table such that the "COUNT animals" query returns a total of 10.
I updated the database, such that the "gabumon is returned when the query is made, as requested.
![Screenshot from 2023-04-01 12-56-15](https://user-images.githubusercontent.com/110289566/229280101-f449a6f4-dfc9-44f3-8f26-02d26f381053.png)

KayLemba reviewed (approved):
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @danielmatama     , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯
- [x] Professional README 💯
- [x] Your work is well documented 🥇 
- [x] Database setup correctly 🥇  











 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



![well-1](https://user-images.githubusercontent.com/64173189/128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c.gif)

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]s

Olamarx reviewed (approved):
Hi @danielmatama, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
<img src="https://media1.giphy.com/media/8jqIQKr6zeh1j3UYhu/giphy.gif"/>
## Highlights
- professional readme file ✔️
- correct setup of database ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
